app/models/minion.rb

```diff
@@ -18,6 +18,8 @@ def assign_role(role:)
       salt.assign_role role: role
     end
     true
+  rescue
```

Comment:
> A generic rescue could be hiding future problems and maybe it's not optimal.

That was my train of thought. I'd prefer white-listing the exceptions that we want to catch. Maybe other exceptions have to be handled differently.

On another note, maybe we can do as we do in [portus](https://github.com/SUSE/Portus/blob/master/lib/portus/http_helpers.rb#L65). That is, to re-route HTTP exceptions into a handler. This way we can also provide better logging.